exist-core/src/main/java/org/exist/xquery/Predicate.java

```diff
@@ -194,9 +194,13 @@ public Sequence evalPredicate(final Sequence outerSequence,
                         innerSeq = inner.eval(Dependency.dependsOn(inner.getDependencies(), Dependency.CONTEXT_ITEM)
                                 ? contextSequence : null);
                     }
-                    if (innerSeq.getCardinality().isSubCardinalityOrEqualOf(Cardinality.EXACTLY_ONE)) {
+
+                    if (innerSeq.getCardinality().equals(Cardinality.EMPTY_SEQUENCE)) {
```

Comment:
in essence I have the same check...... I think.

Reply:
Are you sure `isEmpty()` works the way we think it does? @dizzzz

Reply:
I checked, it is exactly what we want :)